### ISSUE TITLE
Use the same gmp as gap

### ISSUE THREAD

**Issue by csoeger** [gap-packages/NormalizInterface#15]
On my laptop I use the system gmp and I only have to specify --with-gmp=system for the configure of GAP itself.
But on another machine (Linux) with an outdated system gmp I'm using a locally installed gmp version. I specify the path to the GAP configure and the Normaliz cmake. But the NormalizInterface configure does not recognize a --with-gmp flag. In the compilation of NormalizInterface then the system gmp is used and compilation fails with an error
```
In file included from /home/math/csoeger/gap4r7/src/integer.h:19,
                 from /home/math/csoeger/gap4r7/src/compiled.h:33,
                 from src/normaliz.h:29,
                 from src/normaliz.cc:25:
/home/math/csoeger/gap4r7/src/gmpints.h:63:3: error: #error Aborting compile: GAP requires GMP 5.0.2 or newer
```

We should support --with-gmp for the NormalizInterface configure. Or even better, detect it from the GAP config.
It also would be nice to have that for the build_normaliz.sh, but that is a minor point.

**Comment by csoeger** [comment]
A workaround for this problem to add the gmp include dir to the GAP_CPPFLAGS in the Makefile after the configure.

**Comment by csoeger** [comment]
I get the same problem when compiling the package io. So this seems to be a more general problem.

**Comment by fingolfin** [comment]
We definitely should have a --with-gmp. And of course we should also try to extract info about the correct GMP version from GAP. I have the same problem with e.g. the nq package, and will try to implement generic solution.

**Comment by fingolfin** [comment]
I need to correct myself: There is no point (as far as I can tell, at least) in us adding `--with-gmp`, simply because we *must* use the exact same GMP used by GAP, and also `normaliz` must use that. So the thing to worry about is the latter.
As to which GMP to use: If the user has managed to setup both GAP and normaliz in such a way, then they will have no problem to do the same with NormalizInterface. In the worst case, they just need to pass custom CPPFLAGS / LDFLAGS to configure, like so:
```
./configure \
    --with-gaproot=$GAPROOT \
    --with-normaliz=/some/path \
    CPPFLAGS="-I/path/to/gmp/includes" \
    LDFLAGS="-L/path/to/gmp/lib"
```
Now I can still add a `--with-gmp=PREFIX` option for convenience, but I believe the real issue is to help users get the compatible versions of GAP and Normaliz going. I.e. this is a matter of documentation, and closely related to issue #5.

There are multiple scenarios to consider:

1. the user already has a working GAP installation; we then need to tell them how to build a Normaliz version with the right GMP
2. the user already has both a working GAP, and a working Normaliz; we probably then need to tell them to compile a custom libnormaliz anyway, simply to ensure the GMP version is correct (but also because a working Normaliz does not mean that there also is a libnormaliz and headers available, right?)
3. the user wants (?) to compile everything from scratch: simply an extended version of the first point, I guess... Might actually be a tad easier, because we have full control over which GMP is used... *argh*

Extra caveat:
* in order to compile Normaliz, the user needs to have boost installed somehow (e.g. via a package manager). We could (or could not) give hints on how to do that -- but strictly speaking, that's a job for the Normaliz documentation.

**Comment by fingolfin** [comment]
So I just finished the enhancements to `build-normaliz.sh` we discussed on Skype yesterday.

Only to discover a major issue I completely missed before: the GMP compiled by GAP is missing the gmpxx C++ wrappers for GMP. So, we cannot use the GMP shipped with GAP. Users must use a system GMP for GAP and NormalizInterface. 

Or else GAP must compile the C++ wrappers for GMP, too... Argh.

**Comment by fingolfin** [comment]
See also gap-system/gap#399

**Comment by fingolfin** [comment]
This should now be addressed in d598af3f60fcf3e9fb59048bd58f41f72e7c9f71 -- please carefully read the revised README.md, and give the new build / configure process a spin.

I briefly considered writing more about the `GMP_DIR` and `BOOST_ROOT` environment variables in the README (as I have to use them mysefl in order to use the `build_normaliz.sh` script, but ultimately decided against it, as this is likely only useful for pro users. Should we get a lot of support requests involving those, we can still add them.

**Comment by fingolfin** [comment]
I think this is complete now.
